### ISSUE TITLE
CS9203 False Positive

### ISSUE THREAD

**Issue by AliveDevil** [dotnet/roslyn#74854]
**Version Used**: 8.0.400
```
.NET SDK:
 Version:           8.0.400
 Commit:            36fe6dda56
 Workload version:  8.0.400-manifests.56cd0383
 MSBuild version:   17.11.3+0c8610977
```

**Steps to Reproduce**:

```
ReadOnlySpan<string> strings;
if (flag)
{
    strings = ["Test"]; // CS9203
}
else
{
    strings = [];
}
```

[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0ATEBqAPgAQCYBGAWACh8BmAAiJoGEaBvCm9u2/FGgWQApgECABsaAMxEBDAOZoaAIRoA3AJRsOrchx00ASjCkYA8gDsRATwDKABymmAPAoB8NYDQC8NANoBdANwaujQAluI0/JKy6trBLEFx7AD0Se5e3soBNCkMVgCchAAMVAk6AL4JFbHsCdR0PABygsJiUXKKKjE6WokGRmaWtvZOrmkS0jI0APw+mTQgPgGVCbW0sEYQ5hY0AM4ALlAArmB7ignMNFVlQA)

**Diagnostic Id**:

```
error CS9203: A collection expression of type 'ReadOnlySpan<C.B>' cannot be used in this context because it may be exposed outside of the current scope.
```

**Expected Behavior**:
No false-positive, as with
```
ReadOnlySpan<string> strings = flag ? ["Test"] : [];
```

**Actual Behavior**:
False positive is thrown, as the conditional expression emits the same IL as the if-block.

**Comment by AliveDevil** [comment]
Well, didn't know about `scoped ReadOnlySpan<B>`.
